hail/python/test/hailtop/batch/test_batch.py

```diff
@@ -521,3 +542,18 @@ def test_envvar(self):
         j.env('SOME_VARIABLE', '123abcdef')
         j.command('[ $SOME_VARIABLE = "123abcdef" ]')
         assert b.run().status()['state'] == 'success'
+
+    def test_single_job_with_shell(self):
+        msg = 'hello world'
+        b = self.batch()
+        j = b.new_job(shell='/bin/sh')
+        j.command(f'echo "{msg}" > {j.ofile}')
+        assert b.run().status()['state'] == 'success'
```

Comment:
You can use log to get the output, just don't redirect it.  Ideal if the local and service tests match.

Reply:
I found a solution to reading the log, but I suspect not the easiest one. I searched documentation, nothing on logs besides something that would require redirecting echo to file handle, and since neither batch.Job nor batch.Batch have a get-log method, I went through _batch_client.